### ISSUE TITLE
TERM is set to "dumb" since updating to roxterm 3.15.2 from 3.14.3

### ISSUE THREAD

**Issue by twegener-embertec** [realh/roxterm#269]
Since updating to roxterm 3.15.2 (from 3.14.3) the TERM environment variable is set to "dumb".
This results in `WARNING: terminal is not fully functional` messages an not properly functional pagers.

```
$ echo $TERM
dumb

$ rpm -q roxterm
roxterm-3.15.2-1.fc40.x86_64

```

If I run GNOME Terminal, the TERM variable is correctly set to `xterm-256color`.
The TERM environment variable is not set explicitly in any of the shell configuration files.

**Comment by twegener-embertec** [comment]
Perhaps this is related to 21587477e55e0cf59c9b1123537faafc956906d7 ?

**Comment by scimerman** [comment]
Can confirm the same issue here. Had to set in `.bash_profile` the `export TERM=xterm-256color` and then in the roxterm settings > profile > command tick the box `login` in order to bypass this **dumb** variable.

But since that was not the end of it, I downgraded it back from 3.15.2 to 3.14.3.

**Comment by realh** [comment]
Can you test latest master to see whether that fixes it?

Most of 2158747 got reverted, so the relevant code didn't change that much between 3.14.3 and 3.15.2. What I think is significant is that 3.14.3 deletes TERM, because vte's spawn functions set TERM (defaulting to "xterm-256color") if it isn't already present in the incoming environment. That line went missing, so I've restored it. In GNOME on Arch Linux, the incoming environment has no TERM, but apparently it gets set to "dumb" in whatever you're using.

Instead of altering .bash_profile, you could set a TERM override in your roxterm profile(s).

> But since that was not the end of it, I downgraded it back from 3.15.2 to 3.14.3.

Do you mean you were having some other problems?

**Comment by twegener-embertec** [comment]
I tried building from src at 17f3277a63a718a8aac3e3e8bc89c557dc6d4489 (master) and launching the roxterm executable via the file manager, it still resulted in:
```
$ echo $TERM
dumb
```

This is in Fedora 40 / cinnamon desktop.

The workaround of editing the profile settings in roxterm, going to General and setting TERM environment variable to `xterm-256color` appears to be a satisfactory workaround.

**Comment by scimerman** [comment]
> Do you mean you were having some other problems?

Yes, but since I was working on many things at the same time I didn't remember what it was. Otherwise I would already mentioned it here before or opened another bug. Sorry.

**Comment by realh** [comment]
> I tried building from src at [17f3277](https://github.com/realh/roxterm/commit/17f3277a63a718a8aac3e3e8bc89c557dc6d4489) (master) and launching the roxterm executable via the file manager, it still resulted in:
> 
> ```
> $ echo $TERM
> dumb
> ```
> 
> This is in Fedora 40 / cinnamon desktop.
> 
> The workaround of editing the profile settings in roxterm, going to General and setting TERM environment variable to `xterm-256color` appears to be a satisfactory workaround.

Strange. Did you make sure there was no instance of the old version still running? The default behaviour is that there should only be one instance running at a time, so launching it again passes the command on to the existing instance. Also, when compiled from an untagged commit, it should show the truncated commit hash at the end of the version number in Help > About ROXTerm.

**Comment by twegener-embertec** [comment]
> Strange. Did you make sure there was no instance of the old version still running? The default behaviour is that there should only be one instance running at a time, so launching it again passes the command on to the existing instance.

Ah! After removing the profile workaround setting of TERM, then closing existing instances, and running via the file manager the roxterm built from source, gives different behaviour:
```
$ echo $TERM
xterm-256color
```
And Help -> About ROXTerm then gives `RoxTerm 3.15.2.10-g17f3277`.

However, it issues a dialog on startup saying:
`Error from ROXTerm - Unable to read keys from shortcuts file Shortcuts/Default: unknown reason`

Also, it shows the display as white on black rather than my normally configured black on white.
